test/unit/picolibrary/output_stream/main.cc

```diff
@@ -0,0 +1,864 @@
+    EXPECT_TRUE( stream.fatal_error_present() );
+}
+
+/**
+ * \brief Verify picolibrary::Output_Stream::put( std::int8_t ) works properly.
+ */
+TEST( putSignedByte, worksProperly )
+{
+    auto stream = Mock_Output_Stream{};
+
+    auto const value = random<std::int8_t>();
+
+    EXPECT_CALL( stream.buffer(), put( SafeMatcherCast<std::int8_t>( Eq( value ) ) ) )
+        .WillOnce( Return( Result<Void, Error_Code>{} ) );
+
+    EXPECT_FALSE( stream.put( value ).is_error() );
+
+    EXPECT_TRUE( stream.is_nominal() );
+}
+
+/**
+ * \brief Verify picolibrary::Output_Stream::put( std::int8_t const *, std::int8_t const *
+ *        ) properly handles the presence of an I/O error and/or a fatal error.
+ */
+TEST( putSignedByteBlock, errorPresent )
+{
+    auto stream = Mock_Output_Stream{};
+
+    stream.report_random_error();
+
+    auto const io_error_present    = stream.io_error_present();
+    auto const fatal_error_present = stream.fatal_error_present();
+
+    EXPECT_CALL( stream.buffer(), put( A<std::vector<std::int8_t>>() ) ).Times( 0 );
+
+    auto const values = random_container<std::vector<std::int8_t>>();
+    auto const result = stream.put( &*values.begin(), &*values.end() );
+
+    ASSERT_TRUE( result.is_error() );
+    EXPECT_EQ( result.error(), Generic_Error::IO_STREAM_DEGRADED );
+
+    EXPECT_FALSE( stream.end_of_file_reached() );
+    EXPECT_EQ( stream.io_error_present(), io_error_present );
+    EXPECT_EQ( stream.fatal_error_present(), fatal_error_present );
+}
+
+/**
+ * \brief Verify picolibrary::Output_Stream::put( std::int8_t const *, std::int8_t const *
+ *        ) properly handles a put error.
+ */
+TEST( putSignedByteBlock, putError )
+{
+    auto stream = Mock_Output_Stream{};
+
+    auto const error = random<Mock_Error>();
+
+    EXPECT_CALL( stream.buffer(), put( A<std::vector<std::int8_t>>() ) ).WillOnce( Return( error ) );
+
+    auto const values = random_container<std::vector<std::int8_t>>();
+    auto const result = stream.put( &*values.begin(), &*values.end() );
+
+    ASSERT_TRUE( result.is_error() );
+    EXPECT_EQ( result.error(), error );
+
+    EXPECT_FALSE( stream.end_of_file_reached() );
+    EXPECT_FALSE( stream.io_error_present() );
+    EXPECT_TRUE( stream.fatal_error_present() );
+}
+
+/**
+ * \brief Verify picolibrary::Output_Stream::put( std::int8_t const *, std::int8_t const *
+ *        ) works properly.
+ */
+TEST( putSignedByteBlock, worksProperly )
+{
+    auto stream = Mock_Output_Stream{};
+
+    auto const values = random_container<std::vector<std::int8_t>>();
+
+    EXPECT_CALL( stream.buffer(), put( values ) ).WillOnce( Return( Result<Void, Error_Code>{} ) );
+
+    EXPECT_FALSE( stream.put( &*values.begin(), &*values.end() ).is_error() );
+
+    EXPECT_TRUE( stream.is_nominal() );
+}
+
+/**
+ * \brief Verify picolibrary::Output_Stream::print() properly handles the presence of an
+ *        I/O error and/or a fatal error.
+ */
+TEST( print, errorPresent )
```

Comment:
This test (and its name and description) will need to be updated when error handling is revised.